src/riddle/useCases/solveRiddle/__tests__/createSolveRiddleService.test.ts

```diff
@@ -0,0 +1,126 @@
+
+import { describe, it, expect } from 'vitest';
+import { createSolveRiddleModel } from '../createSolveRiddleService';
+
+const mockRiddle = {
+  id: 'riddle-1',
+  contents: 'What has keys but can’t open locks?',
+  answers: [
+    { id: 'a', text: 'Piano' },
+    { id: 'b', text: 'Map' },
+    { id: 'c', text: 'Locksmith' },
+  ],
+};
+
+describe('createSolveRiddleModel', () => {
+  it('returns true for shouldDisplayRiddle ', () => {
+    const model = createSolveRiddleModel({
+      isLoading: false,
+      selected: undefined,
+      correct: undefined,
+      randomRiddle: mockRiddle,
+      answersAvailable: true,
+    });
+    expect(model.shouldDisplayRiddle).toBe(true);
```

Comment:
(optional, ditto throughout) It improves readability when the AAA sections are separated with a blank line:

```suggestion

    expect(model.shouldDisplayRiddle).toBe(true);
```